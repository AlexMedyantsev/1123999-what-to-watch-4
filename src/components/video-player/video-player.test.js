import VideoPlayer from "./video-player.jsx";
import React from "react";
import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {BrowserRouter} from "react-router-dom";
import configureStore from "redux-mock-store";

const movie = {
  preview: `https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4`,
  title: `The Legend`,
  isPlaying: false,
};

const mockStore = configureStore([]);


it(`VideoPlayer is rendered correctly`, () => {
  const {preview, title, isPlaying} = movie;

  const mockFunction = jest.fn();
  const store = mockStore({
  });

  const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <VideoPlayer
            movieLink={preview}
            movieTitle={title}
            isPlaying={isPlaying}
            clickPlayHandler={mockFunction}
            clickPauseHandler={mockFunction}
          />,
        </BrowserRouter>
      </Provider>, {
        createNodeMock: () => {
          return {};
        }
      }
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
