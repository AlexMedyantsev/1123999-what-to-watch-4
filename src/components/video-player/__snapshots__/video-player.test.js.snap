// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer is rendered correctly 1`] = `
Array [
  <div
    className="player"
  >
    <video
      className="player__video"
      poster="img/player-poster.jpg"
      src="https://archive.org/download/BigBuckBunny_124/Content/big_buck_bunny_720p_surround.mp4"
    />
    <button
      className="player__exit"
      onClick={[Function]}
      type="button"
    >
      Exit
    </button>
    <div
      className="player__controls"
    >
      <div
        className="player__controls-row"
      >
        <div
          className="player__time"
        >
          <progress
            className="player__progress"
            max="100"
            value="0"
          />
          <div
            className="player__toggler"
          >
            Toggler
          </div>
        </div>
        <div
          className="player__time-value"
        >
          
        </div>
      </div>
      <div
        className="player__controls-row"
      >
        <button
          className="player__play"
          onClick={[Function]}
          type="button"
        >
          <svg
            fill="none"
            height="19"
            viewBox="0 0 19 19"
            width="19"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M0 0L19 9.5L0 19V0Z"
              fill="#EEE5B5"
              fillRule="evenodd"
            />
          </svg>
          <span>
            Play
          </span>
        </button>
        <div
          className="player__name"
        >
          The Legend
        </div>
        <button
          className="player__full-screen"
          onClick={[Function]}
          type="button"
        >
          <svg
            height="27"
            viewBox="0 0 27 27"
            width="27"
          >
            <use
              xlinkHref="#full-screen"
            />
          </svg>
          <span>
            Full screen
          </span>
        </button>
      </div>
    </div>
  </div>,
  ",",
]
`;
